app/src/main/java/com/example/cinemapp/ui/main/ProfileViewModel.kt

```diff
@@ -1,13 +1,107 @@
 package com.example.cinemapp.ui.main
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.example.cinemapp.data.MovieRepository
 import com.example.cinemapp.data.UserPreferences
+import com.example.cinemapp.ui.main.model.AccountDetails
+import com.example.cinemapp.ui.main.model.MovieCard
+import com.example.cinemapp.util.mappers.ProfileMapper
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
 
 class ProfileViewModel(
     private val movieRepository: MovieRepository,
-    val userPrefs: UserPreferences
+    private val userPrefs: UserPreferences,
+    private val profileMapper: ProfileMapper
 ) : ViewModel() {
-    val session = userPrefs.getSessionId()
+
+    data class State(
+        val accountDetails: AccountDetails? = null,
+        val pagesLoaded: Int = 0,
+        val movies: List<MovieCard> = emptyList(),
+        val movieListType: MovieRepository.MovieListType = MovieRepository.MovieListType.FAVORITE
+    )
+
+    private val _state = MutableStateFlow(State())
+    val state: StateFlow<State> = _state
+    private val _signOut: MutableSharedFlow<Unit> = MutableSharedFlow()
+    val signOut = _signOut.asSharedFlow()
+    private val _notSignedIn: MutableSharedFlow<Unit> = MutableSharedFlow()
+    val notSignedIn = _notSignedIn.asSharedFlow()
+    private var isPaging = false
+    private val session = userPrefs.getSessionId()
+
+
+    fun signOut() {
+        viewModelScope.launch {
+            session.collect { sessionId ->
+                sessionId?.let {
+                    movieRepository.deleteSession(it)
+                }
+                userPrefs.deleteSessionId()
+                _signOut.emit(Unit)
+            }
+        }
+    }
+
+    fun getAccountDetails() {
+        viewModelScope.launch {
+            session.collect { sessionId ->
```

Comment:
Tou can call session.firstOrNull() method on flow to just take a value from the flow, so you do not have to collect the whole flow